distributed/protocol/torch.py

```diff
@@ -16,13 +16,13 @@ def serialize_torch_Tensor(t):
     requires_grad_ = t.requires_grad
 
     if requires_grad_:
-        sub_header, frames = serialize(t.detach().numpy())
+        sub_header, frames = serialize(t.detach().cpu().numpy())
     else:
-        sub_header, frames = serialize(t.numpy())
+        sub_header, frames = serialize(t.cpu().numpy())
 
     header = {"sub-header": sub_header}
     if t.grad is not None:
-        grad_header, grad_frames = serialize(t.grad.numpy())
+        grad_header, grad_frames = serialize(t.grad.cpu().numpy())
         header["grad"] = {"header": grad_header, "start": len(frames)}
         frames += grad_frames
     header["requires_grad"] = requires_grad_
```

Comment:
Not able to leave a suggestion, but looks like we'll also want to add some GPU-specific handling to the the `grad` assignment below:

```patch
diff --git a/distributed/protocol/torch.py b/distributed/protocol/torch.py
index d02da32b..e3afc2e1 100644
--- a/distributed/protocol/torch.py
+++ b/distributed/protocol/torch.py
@@ -49,7 +49,10 @@ def deserialize_torch_Tensor(header, frames):
             data=x, device=header["device"], requires_grad=header["requires_grad"]
         )
     if grad is not None:
-        t.grad = torch.from_numpy(grad)
+        if header["device"] == "cpu":
+            t.grad = torch.from_numpy(grad)
+        else:
+            t.grad = torch.tensor(data=grad, device=header["device"])
     return t
```

This patch got tests passing locally for me